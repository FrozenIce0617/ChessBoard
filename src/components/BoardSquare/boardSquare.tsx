import React, { ReactNode } from 'react';

import Square from '../Square';

interface IProps {
  posX: number;
  posY: number;
  isOverlay: boolean;
  children: ReactNode;
}

class BoardSquare extends React.Component<IProps> {
  render() {
    const { posX, posY, isOverlay, children } = this.props;
    const isBlack = (posX + posY) % 2 === 1;

    return (
      <div
        style={{
          position: 'relative',
          width: '100%',
          height: '100%',
        }}
      >
        <Square isBlack={isBlack} isOverlay={isOverlay}>
          {children}
        </Square>
      </div>
    );
  }
}

export default BoardSquare;
