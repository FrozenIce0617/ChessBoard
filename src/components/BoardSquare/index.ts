import BoardSquare from './boardSquare';

export default BoardSquare;
