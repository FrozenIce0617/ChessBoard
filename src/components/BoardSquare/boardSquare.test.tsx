import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import BoardSquare from './boardSquare';

describe('Square Component', () => {
  it('render component', () => {
    const component = shallow(
      <BoardSquare posX={5} posY={5} isOverlay={true} />,
    );

    expect(shallowToJson(component)).toMatchSnapshot();
  });
});
