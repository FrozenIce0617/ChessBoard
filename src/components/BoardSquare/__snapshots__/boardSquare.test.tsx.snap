// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Square Component render component 1`] = `
<div
  className="squareContainer"
>
  <Square
    isBlack={false}
    isOverlay={true}
  />
</div>
`;
